probe-rs/src/architecture/arm/core/armv6m.rs

```diff
@@ -595,12 +595,15 @@ impl<'probe> CoreInterface for Armv6m<'probe> {
     }
 
     fn reset(&mut self) -> Result<(), Error> {
-        self.sequence.reset_system(&mut self.memory)
+        self.sequence
+            .reset_system(&mut self.memory, crate::CoreType::Armv6m, None)
```

Comment:
I think we should store the `CoreType` in the sequence when we create it. That way we do not have to pass it over and over.
Or is that maybe an issue with dual core systems? I am not sure atm. We can surely merge as is but storing it internally would be cleaner imo.

Reply:
Right now the sequence is created once and cloned for each core, but the address is per core so if it's state on the sequence struct we'd get the wrong value as it's cloned and passed around.

My preference would be to keep this as is since we're already passing per core state to the methods - the MemAP instance.

Reply:
After looking at this some more I do think there's an interesting question of if the sequence should be per SoC, as it is today, or per core.  Per core would be required for some of the rare chips that have RISC-V + ARM on the same SoC (I have a device like that and boy does it confuse debugger software...).

Per core also makes it easier to attach state to the sequence, though we'd still have to borrow the memory on demand to avoid locking the probe out too long.  I think my suggestion is to check this in as is but as more multi-core support comes online revisit the sequence instantiation all up at that time.  I suspect by the time I work on the v8-a Raspberry Pi it'll stretch multi-core support enough to figure out how impactful this is.